accio-tests/src/test/java/io/accio/testing/TestMDLResource.java

```diff
@@ -0,0 +1,156 @@
+            throws Exception
+    {
+        CheckOutputDto startUp = getDeployStatus();
+        assertThat(startUp.getStatus()).isEqualTo(CheckOutputDto.Status.READY);
+        assertThat(startUp.getManifest().getModels().get(0).getColumns().size()).isEqualTo(1);
+        assertThatNoException().isThrownBy(() -> deployMDL(updated));
+        CheckOutputDto afterDeploy = getDeployStatus();
+        waitUntilPrepare();
+        assertThat(afterDeploy.getManifest().getModels().get(0).getColumns().size()).isEqualTo(2);
+        waitUntilReady();
+
+        assertThatNoException().isThrownBy(() -> deployMDL(initial));
+        CheckOutputDto afterDeploy2 = getDeployStatus();
+        waitUntilPrepare();
+        assertThat(afterDeploy2.getManifest().getModels().get(0).getColumns().size()).isEqualTo(1);
+        waitUntilReady();
+
+        assertThat(requireNonNull(mdlDir.resolve("archive").toFile().listFiles()).length).isEqualTo(2);
+    }
+
+    @Test
+    public void testPreview()
+    {
+        Manifest previewManifest = Manifest.builder()
+                .setCatalog("canner-cml")
+                .setSchema("tpch_tiny")
+                .setModels(List.of(
+                        model("Customer", "SELECT * FROM \"canner-cml\".tpch_tiny.customer",
+                                List.of(column("custkey", "integer", null, false, "c_custkey")))))
+                .build();
+
+        PreviewDto testDefaultDto = new PreviewDto(previewManifest, "select custkey from Customer", null);
+        List<Object[]> testDefault = preview(testDefaultDto);
+        assertThat(testDefault.size()).isEqualTo(100);
+        assertThat(testDefault.get(0).length).isEqualTo(1);
+
+        PreviewDto testDefaultDto1 = new PreviewDto(previewManifest, "select custkey from Customer limit 200", null);
+        List<Object[]> preview1 = preview(testDefaultDto1);
+        assertThat(preview1.size()).isEqualTo(100);
+        assertThat(preview1.get(0).length).isEqualTo(1);
+
+        PreviewDto testDefaultDto2 = new PreviewDto(previewManifest, "select custkey from Customer limit 200", 150L);
+        List<Object[]> preview2 = preview(testDefaultDto2);
+        assertThat(preview2.size()).isEqualTo(150);
+        assertThat(preview2.get(0).length).isEqualTo(1);
+
+        assertWebApplicationException(() -> preview(new PreviewDto(previewManifest, "select orderkey from Orders limit 100", null)))
+                .hasErrorMessageMatches(".*Table \"Orders\" must be qualified with a dataset.*");
+    }
+
+    private void waitUntilPrepare()
+            throws ExecutionException, InterruptedException, TimeoutException
+    {
+        CompletableFuture.runAsync(() -> {
+            while (true) {
+                CheckOutputDto checkOutputDto = getDeployStatus();
+                if (checkOutputDto.getStatus() == CheckOutputDto.Status.PREPARE) {
+                    break;
+                }
+                try {
+                    Thread.sleep(1000);
+                }
```

Comment:
This may cause flaky test since the status PREPARE may not stay very long. And doing status check in e2e is not a good idea I thought, status check would be appropriate in unit tests. In e2e we should only assert the final status (READY or FAILED)